### PR TITLE
Added clustering

### PR DESCRIPTION
**Pull request by rebelDi**

### CODE REVIEW

File: back end/clustering.py [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+
+word = "bank"
+part_of_speech = "Noun"
+language = "English"
+number_of_clusters = 2
+
+# building model 
+# Dimensionality of the resulting word vectors.
+#more dimensions mean more traiig them, but more generalized
+num_features = 300
```
Comment (psriraj17) [inline back end/clustering.py:49]: Can we define this variable locally?

File: back end/clustering.py [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+
+# Minimum word count threshold - the smallest set of words we want to recognise 
+min_word_count = 3
```
Comment (psriraj17) [inline back end/clustering.py:52]: can we define the below variables locally? Or is it must for us to have them globally?
min_word_count, context_size, seed &  downsampling

File: back end/clustering.py [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+from sklearn.cluster import KMeans
+
+# Download tokenizers
+# nltk.download("punkt") # pretrained tokenizer
+# nltk.download("stopwords") # remove words like and, the, an, a, of
+
+word = "bank"
+part_of_speech = "Noun"
+language = "English"
+number_of_clusters = 2
```
Comment (psriraj17) [inline back end/clustering.py:44]: can we define this locally?

Comment (psriraj17) [reply back end/clustering.py:44]: Also, shouldn't this be dynamic and entered by the user ?

File: back end/clustering.py [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+#how often to use
+# how frequent the word is, the more it's used the less we want to look at it again
+# pass the most unused words
+downsampling = 1e-3
+
+# Seed for the RNG, to make the results reproducible
+# determenistic, good for debugging
+seed = 1
+
+def get_regex():
```
Comment (psriraj17) [inline back end/clustering.py:71]: can we have a comment on what this function does?

File: back end/clustering.py (hunk 1 above)
Comment (rebelDi) [reply back end/clustering.py:49]: Yes, but it would be better if we left it like that, because this variable is a parameter of configuration, so it would be easier for us to change it later instead of looking for it in the function.

File: back end/clustering.py (hunk 2 above)
Comment (rebelDi) [reply back end/clustering.py:52]: they all are used in one function, so we can push it to just be local variables, will it be easier for us to look for them in a function or just at the top of the code in case of changing them?

File: back end/clustering.py (hunk 3 above)
Comment (rebelDi) [reply back end/clustering.py:44]: We need to have default parameters and it is easier to test if we have those

File: back end/clustering.py (hunk 4 above)
Comment (rebelDi) [reply back end/clustering.py:71]: Yes, will add it in the newer version of branch

File: back end/clustering.py (hunk 2 above)
Comment (psriraj17) [reply back end/clustering.py:52]: fine we can have keep them in the top

**Review by psriraj17 — COMMENTED** [review]
Code looks fine to me.

**Review by psriraj17 — APPROVED** [review]
